src/ng/directive/ngRepeat.js

```diff
@@ -115,10 +116,12 @@
  * <br />
  * <div class="alert alert-warning">
  * Avoid using `track by $index` when the repeated template contains
- * {@link guide/expression#one-time-binding one-time bindings}. In such cases, the `nth` DOM
- * element will always be matched with the `nth` item of the array, so the bindings on that element
- * will not be updated even when the corresponding item changes, essentially causing the view to get
- * out-of-sync with the underlying data.
+ * {@link guide/expression#one-time-binding one-time bindings} or {@link guide/directive#creating-directives directives}. 
+ * In such cases, where the array changes and `track by $index` is in use the `nth` DOM element 
```

Comment:
> Where the array changes

This doesn't look like it specifies what you mean, coz this works fine: https://plnkr.co/edit/dxzgjmE5MEA0bwQdckVS?p=preview aswell as https://plnkr.co/edit/2i0WKKzh1uDlj4kOrBFN?p=preview

Even tho I'm changing the array.

Reply:
https://plnkr.co/edit/ytQ9p6YJOfrDubCGIc2N?p=preview

Click the remove and you will see the the UI is not updated and that the fail becomes green.

Reply:
I knpow, it just that making a simple change to ur plunkr solves the issue: https://plnkr.co/edit/d39OLmANDGv8EuxXxA6k?p=preview

Hence it's not very clear that this is the problem with arrays. (basicly I guess it's immutable vs  chaning the existing array)

Reply:
Interesting, what does $filter use? As this is what we are using that causes this issue.

Reply:
Not sure, coz the first plunk uses the filter pipe and seem to work fine.

Reply:
`$filter` users Array.prototype.filter, which also doesn't manipulate the array but instead returns a new array. See: https://github.com/angular/angular.js/blob/master/src/ng/filter/filter.js#L172